### PR TITLE
Adds support for HashiCorp Vault

### PR DESCRIPTION
**Pull request by ksatirli**
Hello maintainers 👋🏼 - adding this as draft while we get the final bits and pieces in places.

---

### What does this PR do?

This PR adds a reference to the HashiCorp Vault blueprint. 

### Motivation

We love EKS and believe we should make it easier to manage and protect secrets (and sensitive data, in general) in an EKS workflow.

### More

- [X] Yes, I have tested the PR using my local account setup  (Provide any test evidence report under Additional Notes)
- [X] Yes, I have added a new example under [examples](https://github.com/aws-ia/terraform-aws-eks-blueprints/tree/main/examples) to support my PR

> As we are adding an external add-on, our example code lives in the external repository.

- [ ] Yes, I have created another PR for add-ons under [add-ons](https://github.com/aws-samples/eks-blueprints-add-ons) repo (if applicable)

> Not applicable.

- [X] Yes, I have updated the [docs](https://github.com/aws-ia/terraform-aws-eks-blueprints/tree/main/docs) for this feature
- [X] Yes, I ran `pre-commit run -a` with this PR

**Note**: Not all the PRs required examples and docs except a new pattern or add-on added.

### For Moderators
- [ ] E2E Test successfully complete before merge?

### CODE REVIEW

File: docs/add-ons/vault.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
+# HashiCorp Vault
+
+[HashiCorp Vault](https://www.vaultproject.io) brokers and deeply integrates with trusted identities to automate access to secrets, data, and systems.
+
+This add-on is implemented as an external add-on. For detailed documentation and usage of the add-on please refer to the add-on [repository](https://github.com/hashicorp/hashicorp-vault-eks-blueprints-addon).
```
Comment (kcoleman731) [inline docs/add-ons/vault.md:5]: Getting a 404 for this link?

File: docs/add-ons/vault.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+## Example
+
+Checkout the full [example](https://github.com/hashicorp/hashicorp-vault-eks-blueprints-addon/tree/main/blueprints/getting-started).
```
Comment (kcoleman731) [inline docs/add-ons/vault.md:9]: same here.

File: modules/kubernetes-addons/main.tf [hunk 3]
```diff
@@ -233,6 +233,15 @@ module "traefik" {
   addon_context     = local.addon_context
 }
 
+module "vault" {
+  count = var.enable_vault ? 1 : 0
+
+  source  = "hashicorp/hashicorp-vault-eks-addon/aws"
```
Comment (kcoleman731) [inline modules/kubernetes-addons/main.tf:239]: please include a specific tag.

**Review by kcoleman731 — COMMENTED** [review]
Awesome to see this. Thank you for the contribution. Minor comments and some 404s on urls. Assume the repo is not yet public?

**Review by askulkarni2 — COMMENTED** [review]
@ksatirli thanks for the PR. Would love to test this out.

File: docs/add-ons/vault.md (hunk 1 above)
Comment (ksatirli) [reply docs/add-ons/vault.md]: You're correct @kcoleman731, this was intentional at the time.

**Review by vara-bonthu — APPROVED** [review]
Thanks for the PR @ksatirli 👍🏼 
LGTM!

**Comment by ksatirli** [comment]
Thanks @vara-bonthu - we're still working on getting some things on our side squared away before we publish the final (= `0.9.0`) of this add-on to the Terraform Registry.

If y'all could wait with merging this until then, that would be very helpful!

**Comment by ksatirli** [comment]
(marked as draft; which I forgot to do when I opened this PR)

**Comment by ksatirli** [comment]
Hey folks! We've made an initial version of the module code available at the links present in this PR.

We anticipate down-the-road changes and may release newer module versions. 

For now, please give Vault a try with the [included example](https://github.com/hashicorp/terraform-aws-hashicorp-vault-eks-addon/tree/main/blueprints/getting-started).

**Comment by vara-bonthu** [comment]
> Hey folks! We've made an initial version of the module code available at the links present in this PR.
> 
> We anticipate down-the-road changes and may release newer module versions.
> 
> For now, please give Vault a try with the [included example](https://github.com/hashicorp/terraform-aws-hashicorp-vault-eks-addon/tree/main/blueprints/getting-started).

Thanks @ksatirli 

I have tested the Vault Add-on and it worked without any issues 👍🏼 

```shell
 vabonthu>…/terraform/eks 
▶ kubectl get pods -n vault | grep vault
vault-0                                 1/1     Running   0          4m5s
vault-agent-injector-7f6bdfc48d-sct5p   1/1     Running   0          4m5s

```

**Review by askulkarni2 — APPROVED** [review]
@ksatirli thanks for contributing this addon. LGTM!

**Pull request merged**